src/viam/sdk/common/service_helper.cpp

```diff
@@ -0,0 +1,42 @@
+#include <viam/sdk/common/service_helper.hpp>
+
+#include <sstream>
+
+namespace viam {
+namespace sdk {
+
+::grpc::Status ServiceHelperBase::fail(::grpc::StatusCode code, const char* message) const noexcept
+    try {
+    std::ostringstream stream;
```

Comment:
It is a little wasteful that some paths will create multiple `ostringstream` objects along the way, but it is the error path so I'm not too concerned about it.